src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[10];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == values.length) {
+            grow();
+        }
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add element by negative index");
+        } else if (index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is too much");
+        } else if (index == size) {
+            add(value);
+        } else {
+            if (values.length == size) {
+                grow();
+            }
+            T[] newValues = (T[]) new Object[values.length];
+            for (int i = 0; i < index; i++) {
+                newValues[i] = values[i];
+            }
+            newValues[index] = value;
+            for (int i = index; i < size; i++) {
+                newValues[i + 1] = values[i];
+            }
+            size++;
+            values = newValues;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("No such element exists");
+        } else if (index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Can't get element by negative index");
+        }
```

Comment:
Make checks shorter and put these to separate mothod. for example:
```suggestion
private void checkIndex(int index) {
        if (index < 0 || index >= size) {
            throw new ArrayListIndexOutOfBoundsException("index " + index
                    + " is out of bound for length " + size);
        } 
```